### PR TITLE
feat: created thumbnail using expo-video-thumbnail

### PR DESCRIPTION
**Pull request by OtavioStasiak**
## Proposed changes

Create a thumbnail to improve the video component UX.

## Issue(s)	

https://rocketchat.atlassian.net/browse/NATIVE-226

### Basic
- Open the app;
- Open a room that has a video;

### Others
- Media auto download (disable/enable auto download of videos)
- E2EE video
- Tablet

#### Before:

https://github.com/user-attachments/assets/bac0bbbe-814b-4873-8151-f85643ae94a7

#### After:

https://github.com/user-attachments/assets/1be27373-a70d-464f-a2a5-2e4cb48b86b3

#### Before:
[android_before.webm](https://github.com/user-attachments/assets/c3144e25-2103-48e6-8ba3-ca5f60e591ec)

#### After:
[videoPrAndroid.webm](https://github.com/user-attachments/assets/55a70e63-db20-4003-86e1-b84573e10c09)

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [X] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [X] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [X] Lint and unit tests pass locally with my changes
- [X] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [X] I have added necessary documentation (if applicable)
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/containers/message/Components/Attachments/Thumbnail.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	overlay: {
+		flex: 1
+	},
+	image: {
+		width: '100%',
+		height: '100%'
+	},
+	playerIcon: {
+		position: 'absolute',
+		shadowColor: '#000',
```
Comment (diegolmello) [inline app/containers/message/Components/Attachments/Thumbnail.tsx:25]: Don't use fixed colors. Everything should come from themes.

File: app/containers/message/Components/Attachments/Thumbnail.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	container: {
+		flex: 1,
+		alignItems: 'center',
+		justifyContent: 'center'
+	},
+	overlay: {
+		flex: 1
+	},
+	image: {
+		width: '100%',
```
Comment (diegolmello) [inline app/containers/message/Components/Attachments/Thumbnail.tsx:20]: Can you use flex?

File: app/containers/message/Components/Attachments/Thumbnail.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	loading: boolean;
+	uri: string | null;
+};
+
+type ThumbnailProps = {
+	url: string;
+	encrypted?: boolean;
+};
+
+const Thumbnail = ({ url, encrypted = false }: ThumbnailProps) => {
```
Comment (diegolmello) [inline app/containers/message/Components/Attachments/Thumbnail.tsx:44]: You asked my opinion about splitting Thumbnail in a separate file.
I said we should either keep on Video file (it's a small file, so it's ok) or we create a Video folder so `message/Components/Attachments` is organized with one file per feature.

You create `Thumbnail` file without a Video folder, so Attachments now have 
```
- CollapsibleQuote (folder)
- Image (folder) 
- AttachedActions.tsx
- Attachments.tsx
- Audio.tsx
- Reply.tsx
- Video.tsx
- Thumbnail.tsx (this one is only related to Video)
```

**Review by diegolmello — CHANGES_REQUESTED** [review]
Almost there.
Lint is failing.

File: app/containers/message/Components/Attachments/Video.tsx [hunk 4]
```diff
@@ -8,70 +10,108 @@ import I18n from '../../../../i18n';
...
-interface IMessageVideo {
-	file: IAttachment;
-	showAttachment?: (file: IAttachment) => void;
-	getCustomEmoji: TGetCustomEmoji;
-	author?: IUserMessage;
-	style?: StyleProp<TextStyle>[];
-	isReply?: boolean;
-	msg?: string;
-}
+type Image = string | null;
```
Comment (diegolmello) [inline app/containers/message/Components/Attachments/Video.tsx:51]:
```suggestion
type TThumbnailImage = string | null;
```

File: app/containers/message/Components/Attachments/Video.tsx [hunk 5]
```diff
@@ -8,70 +10,108 @@ import I18n from '../../../../i18n';
...
+	image: {
+		width: '100%',
+		height: '100%'
 	},
-	text: {
-		...sharedStyles.textRegular,
-		fontSize: 12
+	playerIcon: {
+		position: 'absolute',
+		shadowColor: themes.light.backdropColor,
```
Comment (diegolmello) [inline app/containers/message/Components/Attachments/Video.tsx:42]: light?

File: app/containers/message/Components/Attachments/Video.tsx [hunk 6]
```diff
@@ -8,70 +10,108 @@ import I18n from '../../../../i18n';
...
+		shadowOpacity: 0.3,
```
Comment (diegolmello) [inline app/containers/message/Components/Attachments/Video.tsx:43]: use theme

**Review by diegolmello — CHANGES_REQUESTED** [review]

File: app/containers/message/Components/Attachments/Video.tsx [hunk 7]
```diff
@@ -8,70 +10,109 @@ import I18n from '../../../../i18n';
...
+			) : (
+				<>
+					<FastImage style={styles.image} resizeMode='cover' source={{ uri: image }} />
+					<CustomIcon
+						name={icon}
+						size={54}
+						color={themes[theme].fontPureWhite}
+						style={[
+							styles.playerIcon,
+							{ shadowColor: themes[theme].backdropColor, shadowOpacity: themes[theme].backdropOpacity }
```
Comment (diegolmello) [inline app/containers/message/Components/Attachments/Video.tsx:97]: Missing shadow or elevation on Android.

**Review by diegolmello — APPROVED** [review]
One last missing change requested and you can merge it.

File: app/containers/message/Components/Attachments/Video.tsx (hunk 7 above)
Comment (OtavioStasiak) [reply app/containers/message/Components/Attachments/Video.tsx]: fixed.

**Pull request merged**